flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisContainer.java

```diff
@@ -68,12 +68,12 @@ public RedisContainer(final JedisSentinelPool sentinelPool) {
 	 */
 	@Override
 	public void close() throws IOException {
-		if (this.jedisPool != null) {
-			this.jedisPool.close();
-		}
-		if (this.jedisSentinelPool != null) {
-			this.jedisSentinelPool.close();
-		}
+		getInstance().close();
```

Comment:
Yes because from the constructors, only one of the `jedisSentinelPool` or `jedisPool` can be non-null value and that is what `getInstance()` returns.